### PR TITLE
feat: Add Welcome View  

### PR DESCRIPTION
**Pull request by gravvy23**
Screenshot 2021-03-29 at 22 37 57
Screenshot 2021-03-29 at 22 38 05

### CODE REVIEW

File: frontend/src/components/Masthead/Search.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}));
+
+export function Search() {
+  const classes = useStyles();
+  return (
+    <div className={classes.search}>
+      <div className={classes.searchIcon}>
+        <SearchIcon />
+      </div>
+      <InputBase
```
Comment (Madlenne) [inline frontend/src/components/Masthead/Search.tsx:47]: Couldn't we use our Input.tsx component here? 🤔

File: frontend/src/domain/WeclomeScreen/Form/Form.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  footer: FooterProps;
+  formElements: Array<InputProps>;
+};
+
+export function Form({ submitTitle, footer, formElements }: FormProps) {
+  const classes = useStyles();
+
+  return (
+    <FinalForm
+      onSubmit={console.log}
```
Comment (Madlenne) [inline frontend/src/domain/WeclomeScreen/Form/Form.tsx:64]: Looks like leftover. If onSubmit is required prop, let's pass `() => {}` for now

**Review by Madlenne — COMMENTED** [review]
Nice work overall! Just few aesthetic things.

**Comment by Madlenne** [comment]
Nice to have: kinda fixed height from a specific breakpoint to not let the menu bar overlap the elements below

image

File: frontend/src/domain/WeclomeScreen/WelcomeScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  color: ${theme.colors.mediumOrchid};
+`;
+
+export function WelcomeScreen({ type }: Props) {
+  return (
+    <section>
+      <Welcome />
+      <Modal
+        title={
+          type === "sign in" ? (
```
[the remark this answers is not in the document]
Comment (gravvy23) [reply frontend/src/domain/WeclomeScreen/WelcomeScreen.tsx:18]: With the TypeScript there's no need to do that. If you'd misspell the type you'd get compilation error instantly and with the IDE intellisence you get the typeahead

File: frontend/src/components/Masthead/Masthead.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    boxShadow: "none",
+    color: "white",
+  },
+}));
+
+export function Masthead() {
+  const classes = useStyles();
+  return (
+    <AppBar position="static" className={classes.appBar}>
+      <Toolbar>
```
[the remark this answers is not in the document]
Comment (gravvy23) [reply frontend/src/components/Masthead/Masthead.tsx:21]: No idea ;) I used the example from material UI docs

File: frontend/src/components/Masthead/Search.tsx (hunk 1 above)
Comment (gravvy23) [reply frontend/src/components/Masthead/Search.tsx:47]: Probably yes, I was to lazy to check

**Comment by gravvy23** [comment]
> Nice to have: kinda fixed height from a specific breakpoint to not let the menu bar overlap the elements below
> 
> image
Yes I was thinking about that, but wasn't sure how to achieve it, hiding welcome text and make the modal attached to bottom? the screen resolution would need to be 700px or lower for this to happen. Maybe it's worth createing a separate story for adjusting the app to smaller screens?

File: frontend/src/domain/WeclomeScreen/WelcomeScreen.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (gravvy23) [reply frontend/src/domain/WeclomeScreen/WelcomeScreen.tsx]: Screenshot 2021-04-02 at 15 40 48

File: frontend/src/components/Masthead/Navigation.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { makeStyles } from "@material-ui/core/styles";
+import Button from "@material-ui/core/Button";
+
+type NavItem = {
+  name: string;
+  href: string;
+};
+
+const useStyles = makeStyles((theme) => ({
```
Comment (Madlenne) [inline frontend/src/components/Masthead/Navigation.tsx:9]: Sorry, it's my fault that I didn't add it to design but let's add border-bottom on active tab: 
image

File: frontend/src/components/Masthead/Search.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  inputRoot: {
+    color: "inherit",
+  },
+  inputInput: {
+    padding: theme.spacing(1.3, 1, 1.3, 0),
+    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
+    transition: theme.transitions.create("width"),
```
Comment (Madlenne) [inline frontend/src/components/Masthead/Search.tsx:32]: Can't see that the transitions on width work

transition_width

File: frontend/src/domain/WeclomeScreen/Form/Form.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    fontSize: "1rem",
+  },
+  submit: {
+    color: "inherit",
+    backgroundColor: appTheme.colors.darkOrchid,
+    boxShadow: "none",
+    height: theme.spacing(6),
+    margin: theme.spacing(4, 1, 0, 1),
+    "&:hover": {
+      backgroundColor: appTheme.colors.darkOrchid,
```
Comment (Madlenne) [inline frontend/src/domain/WeclomeScreen/Form/Form.tsx:23]: Looks like there is no change in colour

File: frontend/src/domain/WeclomeScreen/Form/Form.tsx [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+                    </>
+                  )}
+                </Field>
+              </Grid>
+            ))}
+            <Button
+              type="submit"
+              fullWidth
+              variant="contained"
+              disabled={submitting || pristine}
```
Comment (Madlenne) [inline frontend/src/domain/WeclomeScreen/Form/Form.tsx:84]: Can we overwrite styles for `disabled` state? The text is hardly readable.

image

File: frontend/src/domain/WeclomeScreen/WelcomeScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+import { Welcome } from "./Welcome/Welcome";
+import { Props, mapTypeToFormProps } from "./utils";
+import { theme } from "../../constants";
+
+const Purple = styled.div`
+  padding-left: 0.5rem;
+  color: ${theme.colors.mediumOrchid};
+`;
+
+const mapTypeToTitle = (type: Props["type"]) => {
```
Comment (Madlenne) [inline frontend/src/domain/WeclomeScreen/WelcomeScreen.tsx:12]: Or maybe `utils.js` file would be better place for the function? 🤔

File: frontend/src/components/Masthead/Search.tsx (hunk 6 above)
Comment (gravvy23) [reply frontend/src/components/Masthead/Search.tsx:32]: @Madlenne  maybe I'm missing something. Do we have such a requirement that the app should be responsive? Desktops will not be such small, so are you suggesting we should support mobiles/tablets? If so please update the design to include all device sizes. Anyway I implemented really basic responsiveness, so I expect that all app views implemented in future will be responsive now.

File: frontend/src/domain/WeclomeScreen/Form/Form.tsx (hunk 7 above)
Comment (gravvy23) [reply frontend/src/domain/WeclomeScreen/Form/Form.tsx]: fixed

File: frontend/src/domain/WeclomeScreen/Form/Form.tsx (hunk 8 above)
Comment (gravvy23) [reply frontend/src/domain/WeclomeScreen/Form/Form.tsx:87]: updated the color

File: frontend/src/domain/WeclomeScreen/WelcomeScreen.tsx (hunk 9 above)
Comment (gravvy23) [reply frontend/src/domain/WeclomeScreen/WelcomeScreen.tsx]: moved

File: frontend/src/components/Masthead/Navigation.tsx (hunk 5 above)
Comment (gravvy23) [reply frontend/src/components/Masthead/Navigation.tsx:11]: added

**Comment by gravvy23** [comment]
@Madlenne please review again

**Review by Madlenne — APPROVED** [review]
Nice work 👏

**Pull request merged**